torch/csrc/jit/runtime/static/memory_planner.cpp

```diff
@@ -135,6 +135,62 @@ std::vector<StorageGroup> assignStorageToManagedTensors(
   return managed_tensor_groups;
 }
 
+ManagedStorages::ManagedStorages()
+    : storages_(nullptr), size_(0), capacity_(0) {}
+
+ManagedStorages::~ManagedStorages() {
+  deallocate();
+}
+
+void ManagedStorages::allocate(size_t capacity) {
+  if (storages_ != nullptr) {
+    deallocate();
+  }
+  capacity_ = capacity;
+  storages_ = reinterpret_cast<at::StorageImpl*>(
+      new unsigned char[capacity_ * sizeof(at::StorageImpl)]);
+}
+
+void ManagedStorages::deallocate() {
+  if (storages_ != nullptr) {
+    for (size_t idx = 0; idx < size_; idx++) {
+      storages_[idx].~StorageImpl();
+    }
+    delete[] reinterpret_cast<unsigned char*>(storages_);
+    capacity_ = 0;
+    size_ = 0;
+    storages_ = nullptr;
+  }
+}
+
+at::StorageImpl* ManagedStorages::operator[](size_t idx) {
+  TORCH_INTERNAL_ASSERT(storages_ != nullptr);
+  return &storages_[idx];
+}
+
+const at::StorageImpl* ManagedStorages::operator[](size_t idx) const {
+  TORCH_INTERNAL_ASSERT(storages_ != nullptr);
+  return &storages_[idx];
+}
+
```

Comment:
please put trivial accessors like this (and size and capacity below) in the header so that they can easily be inlined without relying on LTO